### PR TITLE
Add JSON_LENGTH for MySQL

### PR DESCRIPTION
**Pull request by MrMitch**
This adds support for the `JSON_LENGTH` function for MySQL.

https://dev.mysql.com/doc/refman/8.0/en/json-attribute-functions.html#function_json-length

### CODE REVIEW

File: src/Query/Mysql/JsonLength.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+        $parser->match(Lexer::T_CLOSE_PARENTHESIS);
+    }
+
+    public function getSql(SqlWalker $sqlWalker)
+    {
+        $target = $sqlWalker->walkStringPrimary($this->target);
+        $path = $sqlWalker->walkStringPrimary($this->path);
+
+        return sprintf('JSON_LENGTH(%s, %s)', $target, $path);
```
Comment (stevelacey) [inline src/Query/Mysql/JsonLength.php:39]:
```suggestion
        if ($path) {
            return sprintf('JSON_LENGTH(%s, %s)', $target, $path);
        }
        
        return sprintf('JSON_LENGTH(%s)', $target);
```

File: tests/Query/Mysql/JsonLengthTest.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+use DoctrineExtensions\Tests\Query\MysqlTestCase;
+
+class JsonLengthTest extends MysqlTestCase
+{
+    public function testJsonLengthWithoutPath(): void
+    {
+        $this->assertDqlProducesSql(
+            "SELECT JSON_LENGTH('{}') from DoctrineExtensions\Tests\Entities\Blank as blank",
+            "SELECT JSON_LENGTH('{}', '$') AS sclr_0 FROM Blank b0_"
```
Comment (stevelacey) [inline tests/Query/Mysql/JsonLengthTest.php:13]:
```suggestion
            "SELECT JSON_LENGTH('{}') from DoctrineExtensions\Tests\Entities\Blank as blank",
            "SELECT JSON_LENGTH('{}') AS sclr_0 FROM Blank b0_"
```

File: src/Query/Mysql/JsonLength.php [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        $parser->match(Lexer::T_OPEN_PARENTHESIS);
+
+        $this->target = $parser->StringPrimary();
+
+        if ($parser->getLexer()->isNextToken(Lexer::T_COMMA)) {
+            $parser->match(Lexer::T_COMMA);
+
+            $this->path = $parser->StringPrimary();
+        } else {
+            $this->path = '$';
```
Comment (stevelacey) [inline src/Query/Mysql/JsonLength.php:28]:
```suggestion
```

**Review by stevelacey — CHANGES_REQUESTED** [review]
Thanks for your contribution. I think when it's possible to mirror the SQL to the DQL exactly we probably should, see above amends.

**Comment by MrMitch** [comment]
Hi @stevelacey, thank you for your feedback. I've rebased the branch on master (since #355 was merged) and applied the changes you proposed (with a slight variation to prevent an error in `SqlWalker::walkStringPrimary` if `$path` is null).

**Comment by stevelacey** [comment]
@MrMitch thanks!

**Pull request merged**